packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -253,7 +253,11 @@ const MetricForm: FC<MetricFormProps> = ({
       timestampColumn: current.timestampColumn || "",
       tags: current.tags || [],
       projects:
-        edit || duplicate ? current.projects || [] : project ? [project] : [],
+        current.projects || edit || duplicate
```

Comment:
Why was this change necessary? I'm wondering if it's because we have a new context to view this in that's not an 'edit' or 'duplicate' situation. Would checking `source` instead be a possible solution that would be more readable?
```suggestion
        source === 'datasource-detail' || edit || duplicate
```

Reply:
@bttf This change adds logic so if we pass in `current.projects` to the form, we automatically seed the MetricForm with those projects. It was requested that when creates a new metric from the datasourceId page, that metric inherits the datasource's projects.

Checking for `source === 'datasource-detail'` does work here, in lieu of checking for `current.projects`. Updating.

Reply:
Addressed in this [commit](https://github.com/growthbook/growthbook/pull/1412/commits/f1affc08c34294aef3062e7f168a71af13fd0cce).